yaptide/routes/task_routes.py

```diff
@@ -0,0 +1,48 @@
+from flask import request
+from flask_restful import Resource
+
+from yaptide.persistence.database import db
+from yaptide.persistence.models import SimulationModel, TaskModel
+
+from yaptide.routes.utils.response_templates import yaptide_response
+
+
+class TaskUpdate(Resource):
+    """Class responsible for updating tasks"""
+
+    @staticmethod
+    def post():
+        """
+        Method updating task state
+        Structure required by this method to work properly:
+        {
+            "simulation_id": <int>,
+            "task_id": <string>,
+            "update_key": <string>,
+            "update_dict": <dict>
+        }
+        simulation_id and task_id self explanatory
+        """
+        payload_dict: dict = request.get_json(force=True)
+        if {"simulation_id", "task_id", "update_key", "update_dict"} != set(payload_dict.keys()):
+            return yaptide_response(message="Incomplete JSON data", code=400)
```

Comment:
```suggestion
            return yaptide_response(message="Incomplete JSON data", code=400)
```
Maybe add in comment more details, like: "got JSON with XXX keys, expected JSON with YYY keys".